frontend/src/models/index.js

```diff
@@ -0,0 +1,119 @@
+import EventModel from './event';
+import User from './user';
+import { types, flow, applySnapshot, resolveIdentifier } from 'mobx-state-tree';
+import { fetchEvents, postEvent, putEvent, login as loginAPI } from '../apis';
+
+const transformToMap = (arr = []) => {
+  const result = arr.reduce((accumulator, current) => {
+    accumulator[current.id] = current;
+    return accumulator;
+  }, {});
+  return result;
+};
+
+const removeId = item => {
+  const itemClone = { ...item };
+  delete itemClone.id;
+  return itemClone;
+};
+
+const UI = types.model({
+  eventList: types.optional(
+    types.model({
+      fetching: false,
+      fetchError: types.optional(
+        types.union(types.string, types.boolean),
+        false,
+      ),
+    }),
+    {},
+  ),
+});
+
+export const RootModel = types
```

Comment:
Maybe rename this to RootStore, since it's used as a store instead of a model?

Reply:
it is the model. The store is instantiate later form the model.